node/cn/api_tracer.go

```diff
@@ -454,10 +460,23 @@ func (api *PrivateDebugAPI) traceBlock(ctx context.Context, block *types.Block,
 	if config != nil && config.Reexec != nil {
 		reexec = *config.Reexec
 	}
-	statedb, err := api.computeStateDB(parent, reexec)
+
+	var statedb *state.StateDB
+	// If we have the state fully available, use that.
+	statedb, err := api.cn.blockchain.StateAtWithGCLock(parent.Root())
 	if err != nil {
-		return nil, err
+		// If no state is locally available, the desired state will be generated.
+		statedb, err = api.computeStateDB(parent, reexec)
+		if err != nil {
+			return nil, err
+		}
+		logger.Debug("Get stateDB by computeStateDB", "block", block.NumberU64())
+	} else {
+		logger.Debug("Get stateDB from stateCache", "block", block.NumberU64(), "&stateDB", fmt.Sprintf("%p", statedb))
```

Comment:
Is it helpful to leave the pointer value of stateDB?

Reply:
I used this during debugging. Now I don't need. I will remove. Thanks.